lib/internal/fs/watchers.js

```diff
@@ -345,11 +346,12 @@ async function* watch(filename, options = kEmptyObject) {
         });
         error.filename = filename;
         handle.close();
-        reject(error);
+        queue.push(error);
+        resolve();
         return;
       }
-
-      resolve({ eventType, filename });
+      queue.push({ eventType, filename });
```

Comment:
It seems there is a lack of back pressure handling here. But I am not sure what would be the best way to handle it as the source cannot be paused...?

Reply:
The real question is if ACTUAL backpressure is necessary or if it would be enought to throw an error if the queue grows beyond 1eN entries (for any value of N).

Reply:
I’m on that first thing tomorrow morning UTC. Unless I hear otherwise I’ll use the option names you suggested above. `maxQueue` and `overflow` with the values being `”swallow” | “throw”`.

thanks!